prettyPrinters/providers.py

```diff
@@ -266,3 +266,83 @@ def update(self):
     def has_children(self):
         # type: () -> bool
         return True
+
+
+def StdRcSummaryProvider(valobj, dict):
+    # type: (SBValue, dict) -> str
+    assert valobj.GetNumChildren() == 3
+
+    strong = valobj.GetChildMemberWithName("strong").GetValueAsUnsigned()
+    weak = valobj.GetChildMemberWithName("weak").GetValueAsUnsigned()
+    return "strong={}, weak={}".format(strong, weak)
+
+
+class StdRcSyntheticProvider:
+    """Pretty-printer for alloc::rc::Rc<T> and alloc::sync::Arc<T>
+
+    struct Rc<T> { ptr: NonNull<RcBox<T>>, ... }
+    struct NonNull<T> { pointer: NonZero<*const T> }
+    struct NonZero<T>(T)
+    struct RcBox<T> { strong: Cell<usize>, weak: Cell<usize>, value: T }
+    struct Cell<T> { value: UnsafeCell<T> }
+    struct UnsafeCell<T> { value: T }
+
+    struct Arc<T> { ptr: NonNull<ArcInner<T>>, ... }
+    struct NonNull<T> { pointer: NonZero<*const T> }
+    struct NonZero<T>(T)
+    struct ArcInner<T> { strong: atomic::AtomicUsize, weak: atomic::AtomicUsize, data: T }
+    struct AtomicUsize { v: UnsafeCell<usize> }
+    struct UnsafeCell<T> { value: T }
+    """
+
+    def __init__(self, valobj, dict, is_atomic=False):
+        # type: (SBValue, dict, bool) -> StdRcSyntheticProvider
+        self.valobj = valobj
+
+        self.ptr = self.valobj.GetChildMemberWithName("ptr").GetChildMemberWithName("pointer").GetChildAtIndex(0)
+        self.value = self.ptr.GetChildMemberWithName("data" if is_atomic else "value")
+
+        self.strong = self.ptr.GetChildMemberWithName("strong").GetChildAtIndex(0).GetChildMemberWithName("value")
+        self.weak = self.ptr.GetChildMemberWithName("weak").GetChildAtIndex(0).GetChildMemberWithName("value")
+
+        process = valobj.GetProcess()
+        self.endianness = process.GetByteOrder()
+        self.pointer_size = process.GetAddressByteSize()
+        self.count_type = valobj.GetType().GetBasicType(eBasicTypeUnsignedLong)
+
+    def num_children(self):
+        # type: () -> int
+        return 3
+
+    def get_child_index(self, name):
+        # type: (str) -> int
+        if name == "strong":
+            return 0
+        if name == "weak":
+            return 1
+        if name == "value":
+            return 2
+        return -1
+
+    def get_child_at_index(self, index):
+        # type: (int) -> SBValue
+        if index == 0:
+            count = self.strong.GetValueAsSigned()
+            data = SBData.CreateDataFromUInt64Array(self.endianness, self.pointer_size, [count])
+            return self.valobj.CreateValueFromData("strong", data, self.count_type)
+        if index == 1:
+            count = self.weak.GetValueAsSigned()
+            data = SBData.CreateDataFromUInt64Array(self.endianness, self.pointer_size, [count])
+            return self.valobj.CreateValueFromData("weak", data, self.count_type)
```

Comment:
It seems we should move code about `usize` to `SBValue` conversion into a separate function because we already have two similar pieces of code in this PR. Also, there is another one in #3037

Reply:
Probably, it should be a method of some new class because we also need to store process properties too (`endianness` and `pointer_size`)

Reply:
Good idea! I've created `ValueBuilder` for such needs.